test/unit/org/apache/cassandra/cql3/ast/Conditional.java

```diff
@@ -34,32 +41,171 @@ default AbstractType<?> type()
         return BooleanType.instance;
     }
 
+    @Override
+    default Conditional visit(Visitor v)
+    {
+        return v.visit(this);
+    }
+
+    default List<Conditional> simplify()
+    {
+        return Collections.singletonList(this);
+    }
+
+    class Where implements Conditional
```

Comment:
moved this class from its own file to here, so all impl are in one place... its a pattern that I started doing with this package so migrated to keep consistent... many elements are small so their own files seems wasteful